---
title: "count_analysis_P6"
author: "Giulio Sucar Pregnolato"
date: '2018-05-31'
output: html_document
---

## 1. Importing data

# 1.1. libraries
```{r echo=TRUE}
require(data.table)
library(stringr)
library(gsubfn)
```

# 1.2. DMR = differentially methylated regions
```{r echo=TRUE}
DMR<-read.csv("~/Desktop/Giulio/WGBS/cov400/DMR1.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(DMR)[2:4]<-c("DMR60", "DMR61", "DMR67")
```

# 1.3. CGI positions
```{r echo=TRUE}
require(data.table)
CGI <- fread("~/Desktop/Giulio/CGIByRegion.txt", sep = "\t", header= TRUE, stringsAsFactors=FALSE)
```

# 1.4. vcfByRegion
```{r echo=TRUE}
VCF<-list()

VCF[[1]]<-read.csv("~/Desktop/Giulio/VCF/vcfByRegion/60.vcfByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(VCF)[1]<-"vcf60"
names(VCF[[1]])[2]<-"vcf60"
VCF[[2]]<-read.csv("~/Desktop/Giulio/VCF/vcfByRegion/61.vcfByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(VCF)[2]<-"vcf61"
names(VCF[[2]])[2]<-"vcf61"
VCF[[3]]<-read.csv("~/Desktop/Giulio/VCF/vcfByRegion/67.vcfByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(VCF)[3]<-"vcf67"
names(VCF[[3]])[2]<-"vcf67"
```

# 1.5. ChIP
```{r}
ChIP<-list()

ChIP[[1]]<-read.csv("~/Desktop/Giulio/ChIP/60.H3K27ac.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[1]<-"60_H3K27ac"
names(ChIP[[1]])[2]<-"60_H3K27ac"
ChIP[[2]]<-read.csv("~/Desktop/Giulio/ChIP/60.H3K27me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[2]<-"60_H3K27me3"
names(ChIP[[2]])[2]<-"60_H3K27me3"
ChIP[[3]]<-read.csv("~/Desktop/Giulio/ChIP/60.H3K36me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[3]<-"60_H3K36me3"
names(ChIP[[3]])[2]<-"60_H3K36me3"
ChIP[[4]]<-read.csv("~/Desktop/Giulio/ChIP/60.H3K4me1.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(chip)[4]<-"60_H3K4me1"
names(ChIP[[4]])[2]<-"60_H3K4me1"
ChIP[[5]]<-read.csv("~/Desktop/Giulio/ChIP/60.H3K4me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[5]<-"60_H3K4me3"
names(ChIP[[5]])[2]<-"60_H3K4me3"
ChIP[[6]]<-read.csv("~/Desktop/Giulio/ChIP/60.H3K9me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[6]<-"60_H3K9me3"
names(ChIP[[6]])[2]<-"60_H3K9me3"
  
ChIP[[7]]<-read.csv("~/Desktop/Giulio/ChIP/61.H3K27ac.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[7]<-"61_H3K27ac"
names(ChIP[[7]])[2]<-"61_H3K27ac"
ChIP[[8]]<-read.csv("~/Desktop/Giulio/ChIP/61.H3K27me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[8]<-"61_H3K27me3"
names(ChIP[[8]])[2]<-"61_H3K27me3"
ChIP[[9]]<-read.csv("~/Desktop/Giulio/ChIP/61.H3K36me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[9]<-"61_H3K36me3"
names(ChIP[[9]])[2]<-"61_H3K36me3"
ChIP[[10]]<-read.csv("~/Desktop/Giulio/ChIP/61.H3K4me1.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[10]<-"61_H3K4me1"
names(ChIP[[10]])[2]<-"61_H3K4me1"
ChIP[[11]]<-read.csv("~/Desktop/Giulio/ChIP/61.H3K4me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[11]<-"61_H3K4me3"
names(ChIP[[11]])[2]<-"61_H3K4me3"
ChIP[[12]]<-read.csv("~/Desktop/Giulio/ChIP/61.H3K9me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[12]<-"61_H3K9me3"
names(ChIP[[12]])[2]<-"61_H3K9me3"

ChIP[[13]]<-read.csv("~/Desktop/Giulio/ChIP/67.H3K27ac.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[13]<-"67_H3K27ac"
names(ChIP[[13]])[2]<-"67_H3K27ac"
ChIP[[14]]<-read.csv("~/Desktop/Giulio/ChIP/67.H3K27me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[14]<-"67_H3K27me3"
names(ChIP[[14]])[2]<-"67_H3K27me3"
ChIP[[15]]<-read.csv("~/Desktop/Giulio/ChIP/67.H3K36me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[15]<-"67_H3K36me3"
names(ChIP[[15]])[2]<-"67_H3K36me3"
ChIP[[16]]<-read.csv("~/Desktop/Giulio/ChIP/67.H3K4me1.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[16]<-"67_H3K4me1"
names(ChIP[[16]])[2]<-"67_H3K4me1"
ChIP[[17]]<-read.csv("~/Desktop/Giulio/ChIP/67.H3K4me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[17]<-"67_H3K4me3"
names(ChIP[[17]])[2]<-"67_H3K4me3"
ChIP[[18]]<-read.csv("~/Desktop/Giulio/ChIP/67.H3K9me3.chipByRegion.txt", header=TRUE, stringsAsFactors=FALSE, sep="\t")
names(ChIP)[18]<-"67_H3K9me3"
names(ChIP[[18]])[2]<-"67_H3K9me3"
```


## 2. Preparing the comparison platform

# 2.1. Merge VCF, CGI and ChIP-seq with DMR data
```{r}
comparisonP6 <- DMR[,1:(ncol(DMR)-1)]

for (i in 1:length(VCF)){
comparisonP6<-merge(comparisonP6, VCF[[i]], by=1, all.x=T)}
comparisonP6[is.na(comparisonP6)] <- "-"

for (i in 1:length(ChIP)){
comparisonP6<-merge(comparisonP6, ChIP[[i]], by=1, all.x=T)}
comparisonP6<-merge(comparisonP6, CGI, by=1, all.x=T)
comparisonP6[is.na(comparisonP6)] <- 0
```

# 2.2. Export dataset
```{r}
write.table(comparisonP6,"/Users/hirstlab/Desktop/Giulio/comparisonP6.txt",sep="\t",row.names=FALSE)
```


# 3. Count Analysis

# 3.1. Create columns annotating information, to expedite computing  
```{r}
impact<-c("HIGH", "MODERATE", "LOW", "MODIFIER")

#comparisonP6 <- fread("~/Desktop/Giulio/countAnalysis/comparisonP6.txt", sep = "\t", header= TRUE, stringsAsFactors=FALSE)

comparisonP6_annotated <- comparisonP6

#Create unique columns for chromosome, position start and position end. Resort.
comparisonP6_annotated$chr<-sapply(strsplit(comparisonP6_annotated[,chr.posStart.posEnd],":"), `[`, 1)
comparisonP6_annotated$posStart<-as.numeric(sapply(strsplit(comparisonP6_annotated[,chr.posStart.posEnd],":"), `[`, 2))
comparisonP6_annotated$posEnd<-as.numeric(sapply(strsplit(comparisonP6_annotated[,chr.posStart.posEnd],":"), `[`, 3))
comparisonP6_annotated<-comparisonP6_annotated[order(comparisonP6_annotated$chr, comparisonP6_annotated$posStart),]

#Create columns for info on VCF60
#number of VCs
comparisonP6_annotated$vcf60_count<- ifelse(grepl("^\\-$",comparisonP6_annotated$vcf60),0,str_count(comparisonP6_annotated$vcf60,"\\%")+1)
#has indel mutation
comparisonP6_annotated$vcf60_indel<- !grepl("^\\-$", comparisonP6_annotated$vcf60)&!grepl("^[0-9]+\\|[A,T,C,G]\\|[C,T,G,A](\\,[C,G,T,A])*\\|", comparisonP6_annotated$vcf60)
#has SNP mutation
comparisonP6_annotated$vcf60_snp<- grepl("^[0-9]+\\|[A,T,C,G]\\|[C,T,G,A](\\,[C,G,T,A])*\\|", comparisonP6_annotated$vcf60)
#has homozygote
comparisonP6_annotated$vcf60_homo<- grepl("1\\/1",comparisonP6_annotated$vcf60)
#has heterozygote
comparisonP6_annotated$vcf60_het<- grepl("0\\/1",comparisonP6_annotated$vcf60)
#impact on expression
comparisonP6_annotated$vcf60_impact<- sapply(strapply(comparisonP6_annotated$vcf60, "\\(([a-z,A-Z,0-9,\\_,\\+]*)\\|"), function(y) paste(unique(y), collapse=" "))
#coding mutation
comparisonP6_annotated$vcf60_coding<- grepl("\\([^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|CODING\\|", comparisonP6_annotated$vcf60)
#noncoding mutation
comparisonP6_annotated$vcf60_noncoding<- grepl("\\([^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|NON_CODING\\|", comparisonP6_annotated$vcf60)

#Create columns for info on VCF61
#number of VCs
comparisonP6_annotated$vcf61_count<- ifelse(grepl("^\\-$",comparisonP6_annotated$vcf61),0,str_count(comparisonP6_annotated$vcf61,"\\%")+1)
#has indel mutation
comparisonP6_annotated$vcf61_indel<- !grepl("^\\-$", comparisonP6_annotated$vcf61)&!grepl("^[0-9]+\\|[A,T,C,G]\\|[C,T,G,A](\\,[C,G,T,A])*\\|", comparisonP6_annotated$vcf61)
#has SNP mutation
comparisonP6_annotated$vcf61_snp<- grepl("^[0-9]+\\|[A,T,C,G]\\|[C,T,G,A](\\,[C,G,T,A])*\\|", comparisonP6_annotated$vcf61)
#has homozygote
comparisonP6_annotated$vcf61_homo<- grepl("1\\/1",comparisonP6_annotated$vcf61)
#has heterozygote
comparisonP6_annotated$vcf61_het<- grepl("0\\/1",comparisonP6_annotated$vcf61)
#impact on expression
comparisonP6_annotated$vcf61_impact<- sapply(strapply(comparisonP6_annotated$vcf61, "\\(([a-z,A-Z,0-9,\\_,\\+]*)\\|"), function(y) paste(unique(y), collapse=" "))
#coding mutation
comparisonP6_annotated$vcf61_coding<- grepl("\\([^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|CODING\\|", comparisonP6_annotated$vcf61)
#noncoding mutation
comparisonP6_annotated$vcf61_noncoding<- grepl("\\([^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|NON_CODING\\|", comparisonP6_annotated$vcf61)

#Create columns for info on VCF67
#number of VCs
comparisonP6_annotated$vcf67_count<- ifelse(grepl("^\\-$",comparisonP6_annotated$vcf67),0,str_count(comparisonP6_annotated$vcf67,"\\%")+1)
#has indel mutation
comparisonP6_annotated$vcf67_indel<- !grepl("^\\-$", comparisonP6_annotated$vcf67)&!grepl("^[0-9]+\\|[A,T,C,G]\\|[C,T,G,A](\\,[C,G,T,A])*\\|", comparisonP6_annotated$vcf67)
#has SNP mutation
comparisonP6_annotated$vcf67_snp<- grepl("^[0-9]+\\|[A,T,C,G]\\|[C,T,G,A](\\,[C,G,T,A])*\\|", comparisonP6_annotated$vcf67)
#has homozygote
comparisonP6_annotated$vcf67_homo<- grepl("1\\/1",comparisonP6_annotated$vcf67)
#has heterozygote
comparisonP6_annotated$vcf67_het<- grepl("0\\/1",comparisonP6_annotated$vcf67)
#impact on expression
comparisonP6_annotated$vcf67_impact<- sapply(strapply(comparisonP6_annotated$vcf67, "\\(([a-z,A-Z,0-9,\\_,\\+]*)\\|"), function(y) paste(unique(y), collapse=" "))
#coding mutation
comparisonP6_annotated$vcf67_coding<- grepl("\\([^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|CODING\\|", comparisonP6_annotated$vcf67)
#noncoding mutation
comparisonP6_annotated$vcf67_noncoding<- grepl("\\([^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|[^\\|]*\\|NON_CODING\\|", comparisonP6_annotated$vcf67)

write.table(comparisonP6_annotated,"/Users/hirstlab/Desktop/Giulio/countAnalysis/comparisonP6_annotated.txt",sep="\t",row.names=FALSE)
```


# 4. Count analysis of colocalized DMRs, DHMs and VCs

# 4.1. Counting 
```{r}
#comparisonP6_annotated <- fread("~/Desktop/Giulio/countAnalysis/comparisonP6_annotated.txt", sep = "\t", header= TRUE, stringsAsFactors=FALSE)

#absolute counts, as reference to the size of the dataset, allowing for horizontal comparisons with other datasets. Note that P 5 and P 6 are undermutated 
countAnalysis.P6<-list()
#total amount of VCs identified as unique to each sample
countAnalysis.P6[["P6.60.VC.all"]]<- 113636
countAnalysis.P6[["P6.61.VC.all"]]<- 112763
countAnalysis.P6[["P6.67.VC.all"]]<- 156001
#all entries in comparison data tables, which include only DMR chunks
countAnalysis.P6[["P6.all"]]<-  nrow(comparisonP6_annotated)
#amount of entries with CGI flag
countAnalysis.P6[["P6.CGI"]]<- sum(comparisonP6_annotated$is_CGI)
#amount of entries with 1 and -1 DMR values, as well as amount of >.8 values 
countAnalysis.P6[["P6.60.DMR.is1.pos"]]<- sum(comparisonP6_annotated$DMR60==1)
countAnalysis.P6[["P6.61.DMR.is1.pos"]]<- sum(comparisonP6_annotated$DMR61==1)
countAnalysis.P6[["P6.67.DMR.is1.pos"]]<- sum(comparisonP6_annotated$DMR67==1)
countAnalysis.P6[["P6.60.DMR.is1.neg"]]<- sum(comparisonP6_annotated$DMR60==-1)
countAnalysis.P6[["P6.61.DMR.is1.neg"]]<- sum(comparisonP6_annotated$DMR61==-1)
countAnalysis.P6[["P6.67.DMR.is1.neg"]]<- sum(comparisonP6_annotated$DMR67==-1)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos"]]<- sum(comparisonP6_annotated$DMR60>0.8)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos"]]<- sum(comparisonP6_annotated$DMR61>0.8)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos"]]<- sum(comparisonP6_annotated$DMR67>0.8)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg"]]<- sum(comparisonP6_annotated$DMR60<(-0.8))
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg"]]<- sum(comparisonP6_annotated$DMR61<(-0.8))
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg"]]<- sum(comparisonP6_annotated$DMR67<(-0.8))
#amount of entries with 1 and -1 DMR values, as well as amount of >.8 values, in CGI context
countAnalysis.P6[["P6.60.DMR.is1.pos.CGI"]]<- sum(comparisonP6_annotated$DMR60==1 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.61.DMR.is1.pos.CGI"]]<- sum(comparisonP6_annotated$DMR61==1 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.67.DMR.is1.pos.CGI"]]<- sum(comparisonP6_annotated$DMR67==1 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.60.DMR.is1.neg.CGI"]]<- sum(comparisonP6_annotated$DMR60==-1 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.61.DMR.is1.neg.CGI"]]<- sum(comparisonP6_annotated$DMR61==-1 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.67.DMR.is1.neg.CGI"]]<- sum(comparisonP6_annotated$DMR67==-1 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.CGI"]]<- sum(comparisonP6_annotated$DMR60>0.8 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.CGI"]]<- sum(comparisonP6_annotated$DMR61>0.8 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.CGI"]]<- sum(comparisonP6_annotated$DMR67>0.8 & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.CGI"]]<- sum(comparisonP6_annotated$DMR60<(-0.8) & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.CGI"]]<- sum(comparisonP6_annotated$DMR61<(-0.8) & comparisonP6_annotated$is_CGI==1)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.CGI"]]<- sum(comparisonP6_annotated$DMR67<(-0.8) & comparisonP6_annotated$is_CGI==1)

#amount of VCs and amount of regions with VCs for each sample
countAnalysis.P6[["P6.60.VC"]]<- sum(comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.VC"]]<- sum(comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.VC"]]<- sum(comparisonP6_annotated$vcf67_count!=0)
countAnalysis.P6[["P6.60.VC.count"]]<- sum(comparisonP6_annotated$vcf60_count)
countAnalysis.P6[["P6.61.VC.count"]]<- sum(comparisonP6_annotated$vcf61_count)
countAnalysis.P6[["P6.67.VC.count"]]<- sum(comparisonP6_annotated$vcf67_count)
```

   #amount of regions with VCs of each biotype
   countAnalysis.P6[["P6.60.VC.biotype"]]<- colSums(sapply(biotype, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_biotype)))
   countAnalysis.P6[["P6.61.VC.biotype"]]<- colSums(sapply(biotype, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_biotype)))
   countAnalysis.P6[["P6.67.VC.biotype"]]<- colSums(sapply(biotype, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_biotype)))
   #amount of regions with VCs of each gene effect
   countAnalysis.P6[["P6.60.VC.effect"]]<- colSums(sapply(effect, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_effect)))
   countAnalysis.P6[["P6.61.VC.effect"]]<- colSums(sapply(effect, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_effect)))
   countAnalysis.P6[["P6.67.VC.effect"]]<- colSums(sapply(effect, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_effect)))
   
```{r}
   #amount of regions with VCs of each impact
   countAnalysis.P6[["P6.60.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)))
   countAnalysis.P6[["P6.61.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)))
   countAnalysis.P6[["P6.67.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)))
   #amount of regions with indel VCs
   countAnalysis.P6[["P6.60.VC.indel"]]<- sum(comparisonP6_annotated$vcf60_indel) 
   countAnalysis.P6[["P6.61.VC.indel"]]<- sum(comparisonP6_annotated$vcf61_indel)
   countAnalysis.P6[["P6.67.VC.indel"]]<- sum(comparisonP6_annotated$vcf67_indel)
   #amount of regions with SNP VCs
   countAnalysis.P6[["P6.60.VC.snp"]]<- sum(comparisonP6_annotated$vcf60_snp) 
   countAnalysis.P6[["P6.61.VC.snp"]]<- sum(comparisonP6_annotated$vcf61_snp)
   countAnalysis.P6[["P6.67.VC.snp"]]<- sum(comparisonP6_annotated$vcf67_snp)
   #amount of regions with homozygous VCs
   countAnalysis.P6[["P6.60.VC.homo"]]<- sum(comparisonP6_annotated$vcf60_homo) 
   countAnalysis.P6[["P6.61.VC.homo"]]<- sum(comparisonP6_annotated$vcf61_homo)
   countAnalysis.P6[["P6.67.VC.homo"]]<- sum(comparisonP6_annotated$vcf67_homo)
   #amount of regions with heterozygous VCs
   countAnalysis.P6[["P6.60.VC.het"]]<- sum(comparisonP6_annotated$vcf60_het) 
   countAnalysis.P6[["P6.61.VC.het"]]<- sum(comparisonP6_annotated$vcf61_het)
   countAnalysis.P6[["P6.67.VC.het"]]<- sum(comparisonP6_annotated$vcf67_het)
   #amount of regions with known coding VCs
   countAnalysis.P6[["P6.60.VC.coding"]]<- sum(comparisonP6_annotated$vcf60_coding) 
   countAnalysis.P6[["P6.61.VC.coding"]]<- sum(comparisonP6_annotated$vcf61_coding)
   countAnalysis.P6[["P6.67.VC.coding"]]<- sum(comparisonP6_annotated$vcf67_coding)
   #amount of regions with known non-coding VCs
   countAnalysis.P6[["P6.60.VC.noncoding"]]<- sum(comparisonP6_annotated$vcf60_noncoding) 
   countAnalysis.P6[["P6.61.VC.noncoding"]]<- sum(comparisonP6_annotated$vcf61_noncoding)
   countAnalysis.P6[["P6.67.VC.noncoding"]]<- sum(comparisonP6_annotated$vcf67_noncoding)
   
#amount of regions with histone mod signals
countAnalysis.P6[["P6.60.HM.active"]]<- sum(comparisonP6_annotated$X60_H3K27ac | comparisonP6_annotated$X60_H3K36me3 | comparisonP6_annotated$X60_H3K4me1 | comparisonP6_annotated$X60_H3K4me3)
countAnalysis.P6[["P6.60.HM.repressive"]]<- sum(comparisonP6_annotated$X60_H3K27me3 | comparisonP6_annotated$X60_H3K9me3)
countAnalysis.P6[["P6.60.HM.H3K27ac"]]<- sum(comparisonP6_annotated$X60_H3K27ac)
countAnalysis.P6[["P6.60.HM.H3K27me3"]]<- sum(comparisonP6_annotated$X60_H3K27me3)
countAnalysis.P6[["P6.60.HM.H3K36me3"]]<- sum(comparisonP6_annotated$X60_H3K36me3)
countAnalysis.P6[["P6.60.HM.H3K4me1"]]<- sum(comparisonP6_annotated$X60_H3K4me1)
countAnalysis.P6[["P6.60.HM.H3K4me3"]]<- sum(comparisonP6_annotated$X60_H3K4me3)
countAnalysis.P6[["P6.60.HM.H3K9me3"]]<- sum(comparisonP6_annotated$X60_H3K9me3)
countAnalysis.P6[["P6.61.HM.active"]]<- sum(comparisonP6_annotated$X61_H3K27ac | comparisonP6_annotated$X61_H3K36me3 | comparisonP6_annotated$X61_H3K4me1 | comparisonP6_annotated$X61_H3K4me3)
countAnalysis.P6[["P6.61.HM.repressive"]]<- sum(comparisonP6_annotated$X61_H3K27me3 | comparisonP6_annotated$X61_H3K9me3)
countAnalysis.P6[["P6.61.HM.H3K27ac"]]<- sum(comparisonP6_annotated$X61_H3K27ac)
countAnalysis.P6[["P6.61.HM.H3K27me3"]]<- sum(comparisonP6_annotated$X61_H3K27me3)
countAnalysis.P6[["P6.61.HM.H3K36me3"]]<- sum(comparisonP6_annotated$X61_H3K36me3)
countAnalysis.P6[["P6.61.HM.H3K4me1"]]<- sum(comparisonP6_annotated$X61_H3K4me1)
countAnalysis.P6[["P6.61.HM.H3K4me3"]]<- sum(comparisonP6_annotated$X61_H3K4me3)
countAnalysis.P6[["P6.61.HM.H3K9me3"]]<- sum(comparisonP6_annotated$X61_H3K9me3)
countAnalysis.P6[["P6.67.HM.active"]]<- sum(comparisonP6_annotated$X67_H3K27ac | comparisonP6_annotated$X67_H3K36me3 | comparisonP6_annotated$X67_H3K4me1 | comparisonP6_annotated$X67_H3K4me3)
countAnalysis.P6[["P6.67.HM.repressive"]]<- sum(comparisonP6_annotated$X67_H3K27me3 | comparisonP6_annotated$X67_H3K9me3)
countAnalysis.P6[["P6.67.HM.H3K27ac"]]<- sum(comparisonP6_annotated$X67_H3K27ac)
countAnalysis.P6[["P6.67.HM.H3K27me3"]]<- sum(comparisonP6_annotated$X67_H3K27me3)
countAnalysis.P6[["P6.67.HM.H3K36me3"]]<- sum(comparisonP6_annotated$X67_H3K36me3)
countAnalysis.P6[["P6.67.HM.H3K4me1"]]<- sum(comparisonP6_annotated$X67_H3K4me1)
countAnalysis.P6[["P6.67.HM.H3K4me3"]]<- sum(comparisonP6_annotated$X67_H3K4me3)
countAnalysis.P6[["P6.67.HM.H3K9me3"]]<- sum(comparisonP6_annotated$X67_H3K9me3)


#Colocalization measurements between DMRs and VCs
#amount of DMRs with VCs for each sample. DMR defined as value 1 or 0.8.
countAnalysis.P6[["P6.60.DMR.is1.pos.VC"]]<- sum(comparisonP6_annotated$DMR60==1 & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.is1.pos.VC"]]<- sum(comparisonP6_annotated$DMR61==1 & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.is1.pos.VC"]]<- sum(comparisonP6_annotated$DMR67==1 & comparisonP6_annotated$vcf67_count!=0)
countAnalysis.P6[["P6.60.DMR.is1.neg.VC"]]<- sum(comparisonP6_annotated$DMR60==-1 & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.is1.neg.VC"]]<- sum(comparisonP6_annotated$DMR61==-1 & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.is1.neg.VC"]]<- sum(comparisonP6_annotated$DMR67==-1 & comparisonP6_annotated$vcf67_count!=0)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.VC"]]<- sum(comparisonP6_annotated$DMR60>0.8 & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.VC"]]<- sum(comparisonP6_annotated$DMR61>0.8 & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.VC"]]<- sum(comparisonP6_annotated$DMR67>0.8 & comparisonP6_annotated$vcf67_count!=0)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.VC"]]<- sum(comparisonP6_annotated$DMR60<(-0.8) & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.VC"]]<- sum(comparisonP6_annotated$DMR61<(-0.8) & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.VC"]]<- sum(comparisonP6_annotated$DMR67<(-0.8) & comparisonP6_annotated$vcf67_count!=0)
#amount of DMRs with VCs for each sample in CGI context. DMR defined as value 1 or 0.8.
countAnalysis.P6[["P6.60.DMR.is1.pos.VC.CGI"]]<- sum(comparisonP6_annotated$DMR60==1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.is1.pos.VC.CGI"]]<- sum(comparisonP6_annotated$DMR61==1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.is1.pos.VC.CGI"]]<- sum(comparisonP6_annotated$DMR67==1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_count!=0)
countAnalysis.P6[["P6.60.DMR.is1.neg.VC.CGI"]]<- sum(comparisonP6_annotated$DMR60==-1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.is1.neg.VC.CGI"]]<- sum(comparisonP6_annotated$DMR61==-1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.is1.neg.VC.CGI"]]<- sum(comparisonP6_annotated$DMR67==-1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_count!=0)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.VC.CGI"]]<- sum(comparisonP6_annotated$DMR60>0.8 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.VC.CGI"]]<- sum(comparisonP6_annotated$DMR61>0.8 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.VC.CGI"]]<- sum(comparisonP6_annotated$DMR67>0.8 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_count!=0)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.VC.CGI"]]<- sum(comparisonP6_annotated$DMR60<(-0.8) & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_count!=0)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.VC.CGI"]]<- sum(comparisonP6_annotated$DMR61<(-0.8) & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_count!=0)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.VC.CGI"]]<- sum(comparisonP6_annotated$DMR67<(-0.8) & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_count!=0)
#amount of DMRs with VCs for each sample, split by impact of mutation. DMR defined as value 1 or 0.8.
countAnalysis.P6[["P6.60.DMR.is1.pos.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60==1)
countAnalysis.P6[["P6.61.DMR.is1.pos.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61==1)
countAnalysis.P6[["P6.67.DMR.is1.pos.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67==1)
countAnalysis.P6[["P6.60.DMR.is1.neg.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60==-1)
countAnalysis.P6[["P6.61.DMR.is1.neg.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61==-1)
countAnalysis.P6[["P6.67.DMR.is1.neg.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67==-1)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60>0.8)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61>0.8)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67>0.8)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60<(-0.8))
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61<(-0.8))
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.VC.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67<(-0.8))

#Colocalization measurements between DMRs and VCs in SNP context, to reduce false positives
#amount of DMRs with VCs for each sample. DMR defined as value 1 or 0.8.
countAnalysis.P6[["P6.60.DMR.is1.pos.VC.snp"]]<- sum(comparisonP6_annotated$DMR60==1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.is1.pos.VC.snp"]]<- sum(comparisonP6_annotated$DMR61==1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.is1.pos.VC.snp"]]<- sum(comparisonP6_annotated$DMR67==1 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.is1.neg.VC.snp"]]<- sum(comparisonP6_annotated$DMR60==-1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.is1.neg.VC.snp"]]<- sum(comparisonP6_annotated$DMR61==-1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.is1.neg.VC.snp"]]<- sum(comparisonP6_annotated$DMR67==-1 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.VC.snp"]]<- sum(comparisonP6_annotated$DMR60>0.8 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.VC.snp"]]<- sum(comparisonP6_annotated$DMR61>0.8 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.VC.snp"]]<- sum(comparisonP6_annotated$DMR67>0.8 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.VC.snp"]]<- sum(comparisonP6_annotated$DMR60<(-0.8) & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.VC.snp"]]<- sum(comparisonP6_annotated$DMR61<(-0.8) & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.VC.snp"]]<- sum(comparisonP6_annotated$DMR67<(-0.8) & comparisonP6_annotated$vcf67_snp)
#amount of DMRs with VCs for each sample in CGI context. DMR defined as value 1 or 0.8.
countAnalysis.P6[["P6.60.DMR.is1.pos.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR60==1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.is1.pos.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR61==1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.is1.pos.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR67==1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.is1.neg.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR60==-1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.is1.neg.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR61==-1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.is1.neg.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR67==-1 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR60>0.8 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR61>0.8 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR67>0.8 & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR60<(-0.8) & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR61<(-0.8) & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.VC.snp.CGI"]]<- sum(comparisonP6_annotated$DMR67<(-0.8) & comparisonP6_annotated$is_CGI==1 & comparisonP6_annotated$vcf67_snp)
#amount of DMRs with VCs for each sample, split by impact of mutation. DMR defined as value 1 or 0.8.
countAnalysis.P6[["P6.60.DMR.is1.pos.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60==1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.is1.pos.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61==1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.is1.pos.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67==1 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.is1.neg.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60==-1 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.is1.neg.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61==-1 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.is1.neg.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67==-1 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60>0.8 & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61>0.8 & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67>0.8 & comparisonP6_annotated$vcf67_snp)
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf60_impact)) & comparisonP6_annotated$DMR60<(-0.8) & comparisonP6_annotated$vcf60_snp)
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf61_impact)) & comparisonP6_annotated$DMR61<(-0.8) & comparisonP6_annotated$vcf61_snp)
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.VC.snp.impact"]]<- colSums(sapply(impact, function(x) grepl(sprintf('\\b%s\\b', x), comparisonP6_annotated$vcf67_impact)) & comparisonP6_annotated$DMR67<(-0.8) & comparisonP6_annotated$vcf67_snp)

#Colocalization measurements between DMRs and HMs
countAnalysis.P6[["P6.60.DMR.is1.pos.HM.active"]]<- sum(comparisonP6_annotated$DMR60==1 & as.logical(comparisonP6_annotated$X60_H3K27ac | comparisonP6_annotated$X60_H3K36me3 | comparisonP6_annotated$X60_H3K4me1 | comparisonP6_annotated$X60_H3K4me3))
countAnalysis.P6[["P6.60.DMR.is1.pos.HM.repressive"]]<- sum(comparisonP6_annotated$DMR60==1 & as.logical(comparisonP6_annotated$X60_H3K27me3 | comparisonP6_annotated$X60_H3K9me3))
countAnalysis.P6[["P6.61.DMR.is1.pos.HM.active"]]<- sum(comparisonP6_annotated$DMR61==1 & as.logical(comparisonP6_annotated$X61_H3K27ac | comparisonP6_annotated$X61_H3K36me3 | comparisonP6_annotated$X61_H3K4me1 | comparisonP6_annotated$X61_H3K4me3))
countAnalysis.P6[["P6.61.DMR.is1.pos.HM.repressive"]]<- sum(comparisonP6_annotated$DMR61==1 & as.logical(comparisonP6_annotated$X61_H3K27me3 | comparisonP6_annotated$X61_H3K9me3))
countAnalysis.P6[["P6.67.DMR.is1.pos.HM.active"]]<- sum(comparisonP6_annotated$DMR67==1 & as.logical(comparisonP6_annotated$X67_H3K27ac | comparisonP6_annotated$X67_H3K36me3 | comparisonP6_annotated$X67_H3K4me1 | comparisonP6_annotated$X67_H3K4me3))
countAnalysis.P6[["P6.67.DMR.is1.pos.HM.repressive"]]<- sum(comparisonP6_annotated$DMR67==1 & as.logical(comparisonP6_annotated$X67_H3K27me3 | comparisonP6_annotated$X67_H3K9me3))
countAnalysis.P6[["P6.60.DMR.is1.neg.HM.active"]]<- sum(comparisonP6_annotated$DMR60==-1 & as.logical(comparisonP6_annotated$X60_H3K27ac | comparisonP6_annotated$X60_H3K36me3 | comparisonP6_annotated$X60_H3K4me1 | comparisonP6_annotated$X60_H3K4me3))
countAnalysis.P6[["P6.60.DMR.is1.neg.HM.repressive"]]<- sum(comparisonP6_annotated$DMR60==-1 & as.logical(comparisonP6_annotated$X60_H3K27me3 | comparisonP6_annotated$X60_H3K9me3))
countAnalysis.P6[["P6.61.DMR.is1.neg.HM.active"]]<- sum(comparisonP6_annotated$DMR61==-1 & as.logical(comparisonP6_annotated$X61_H3K27ac | comparisonP6_annotated$X61_H3K36me3 | comparisonP6_annotated$X61_H3K4me1 | comparisonP6_annotated$X61_H3K4me3))
countAnalysis.P6[["P6.61.DMR.is1.neg.HM.repressive"]]<- sum(comparisonP6_annotated$DMR61==-1 & as.logical(comparisonP6_annotated$X61_H3K27me3 | comparisonP6_annotated$X61_H3K9me3))
countAnalysis.P6[["P6.67.DMR.is1.neg.HM.active"]]<- sum(comparisonP6_annotated$DMR67==-1 & as.logical(comparisonP6_annotated$X67_H3K27ac | comparisonP6_annotated$X67_H3K36me3 | comparisonP6_annotated$X67_H3K4me1 | comparisonP6_annotated$X67_H3K4me3))
countAnalysis.P6[["P6.67.DMR.is1.neg.HM.repressive"]]<- sum(comparisonP6_annotated$DMR67==-1 & as.logical(comparisonP6_annotated$X67_H3K27me3 | comparisonP6_annotated$X67_H3K9me3))
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.HM.active"]]<- sum(comparisonP6_annotated$DMR60>0.8 & as.logical(comparisonP6_annotated$X60_H3K27ac | comparisonP6_annotated$X60_H3K36me3 | comparisonP6_annotated$X60_H3K4me1 | comparisonP6_annotated$X60_H3K4me3))
countAnalysis.P6[["P6.60.DMR.isbigger0.8.pos.HM.repressive"]]<- sum(comparisonP6_annotated$DMR60>0.8 & as.logical(comparisonP6_annotated$X60_H3K27me3 | comparisonP6_annotated$X60_H3K9me3))
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.HM.active"]]<- sum(comparisonP6_annotated$DMR61>0.8 & as.logical(comparisonP6_annotated$X61_H3K27ac | comparisonP6_annotated$X61_H3K36me3 | comparisonP6_annotated$X61_H3K4me1 | comparisonP6_annotated$X61_H3K4me3))
countAnalysis.P6[["P6.61.DMR.isbigger0.8.pos.HM.repressive"]]<- sum(comparisonP6_annotated$DMR61>0.8 & as.logical(comparisonP6_annotated$X61_H3K27me3 | comparisonP6_annotated$X61_H3K9me3))
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.HM.active"]]<- sum(comparisonP6_annotated$DMR67>0.8 & as.logical(comparisonP6_annotated$X67_H3K27ac | comparisonP6_annotated$X67_H3K36me3 | comparisonP6_annotated$X67_H3K4me1 | comparisonP6_annotated$X67_H3K4me3))
countAnalysis.P6[["P6.67.DMR.isbigger0.8.pos.HM.repressive"]]<- sum(comparisonP6_annotated$DMR67>0.8 & as.logical(comparisonP6_annotated$X67_H3K27me3 | comparisonP6_annotated$X67_H3K9me3))
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.HM.active"]]<- sum(comparisonP6_annotated$DMR60<(-0.8) & as.logical(comparisonP6_annotated$X60_H3K27ac | comparisonP6_annotated$X60_H3K36me3 | comparisonP6_annotated$X60_H3K4me1 | comparisonP6_annotated$X60_H3K4me3))
countAnalysis.P6[["P6.60.DMR.isbigger0.8.neg.HM.repressive"]]<- sum(comparisonP6_annotated$DMR60<(-0.8) & as.logical(comparisonP6_annotated$X60_H3K27me3 | comparisonP6_annotated$X60_H3K9me3))
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.HM.active"]]<- sum(comparisonP6_annotated$DMR61<(-0.8) & as.logical(comparisonP6_annotated$X61_H3K27ac | comparisonP6_annotated$X61_H3K36me3 | comparisonP6_annotated$X61_H3K4me1 | comparisonP6_annotated$X61_H3K4me3))
countAnalysis.P6[["P6.61.DMR.isbigger0.8.neg.HM.repressive"]]<- sum(comparisonP6_annotated$DMR61<(-0.8) & as.logical(comparisonP6_annotated$X61_H3K27me3 | comparisonP6_annotated$X61_H3K9me3))
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.HM.active"]]<- sum(comparisonP6_annotated$DMR67<(-0.8) & as.logical(comparisonP6_annotated$X67_H3K27ac | comparisonP6_annotated$X67_H3K36me3 | comparisonP6_annotated$X67_H3K4me1 | comparisonP6_annotated$X67_H3K4me3))
countAnalysis.P6[["P6.67.DMR.isbigger0.8.neg.HM.repressive"]]<- sum(comparisonP6_annotated$DMR67<(-0.8) & as.logical(comparisonP6_annotated$X67_H3K27me3 | comparisonP6_annotated$X67_H3K9me3))

#exporting results
write.table(unlist(countAnalysis.P6), '/Users/hirstlab/Desktop/Giulio/countAnalysis/count_analysis_P6.csv', append= T, sep=',', row.names=TRUE)
```


