### 2018Epigenomics
Late upload of work done in 2018 for the Hirst lab at UBC.

## Data types:
# Genomic data
- WGS: Whole genome sequencing raw data.
- VCF: Variant Call Format. Lists all variations found between a WGS file and a reference human genome, like hg38 or hg19
- 

# Methylomic data 
- WGBS: Whole genome bisulfite sequencing. The bisulfite treatment modifies Cytosines that were unmethylated, so one can extrapolate methylation data from WGBS by comparing with WGS data from the same sample.
- Compare samples to identify differentially methylated regions (DMRs)

# Hystone Modification data
- ChIP-seq 
- Compare samples to find differential histone modifications (DHMs)


# Gene Expression data
- RNA-seq and miRNA-seq: 
- Compare samples to find differentially expressed regions (DEs)


## Data source:
Canadian Epigenomes dataset
http://www.epigenomes.ca/site-data
Colorectal samples, normal and diseased
![alt text](https://github.com/GiulioSP/2018Epigenomics/blob/main/data_breakdown.jpg?raw=true)


## Glossary:
- hg38 -> reference sequence for the human genome, build version 38
- 
